lib/matplotlib/cm.py

```diff
@@ -255,10 +254,18 @@ def to_rgba(self, x, alpha=None, bytes=False, norm=True):
                     xx = x
                 else:
                     raise ValueError("third dimension must be 3 or 4")
-                if bytes and xx.dtype != np.uint8:
-                    xx = (xx * 255).astype(np.uint8)
-                if not bytes and xx.dtype == np.uint8:
-                    xx = xx.astype(float) / 255
+                if xx.dtype.kind == 'f':
+                    if xx.max() > 1 or xx.min() < 0:
```

Comment:
How does this interact with your image refactor?  I think that if this check is executed only on external input of 3-D RGBA arrays, then it is OK; that is really its purpose here.  But if it is executed on internally-generated arrays in the normal course of events, then it needs to be bypassed.  In fact, if that is happening, perhaps it is an entire call to the `to_rgba` method that needs to be bypassed, and the check can stay in place.